### PR TITLE
[SYSTEMML-1863][WIP] Add MLContext test class for linear regression

### PR DESCRIPTION
**Pull request by j143**
This plans to add a test class for all the operations and functions in `LinearRegCG.dml` and `LinearRegDS.dml` scripts.

### CODE REVIEW

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/2026/<h2>Failed Tests: <span class='status-failure'>4</span></h2><h3><a name='SystemML-PullRequestBuilder/org.apache.systemml:systemml' /><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/2026/org.apache.systemml$systemml/testReport'>SystemML-PullRequestBuilder/org.apache.systemml:systemml</a>: <span class='status-failure'>4</span></h3><ul><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/2026/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.mlcontext.algorithms/MLContextLinregTest/testLinregCGDense/'><strong>org.apache.sysml.test.integration.mlcontext.algorithms.MLContextLinregTest.testLinregCGDense</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/2026/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.mlcontext.algorithms/MLContextLinregTest/testLinregDSSparse/'><strong>org.apache.sysml.test.integration.mlcontext.algorithms.MLContextLinregTest.testLinregDSSparse</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/2026/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.mlcontext.algorithms/MLContextLinregTest/testLinregDSDense/'><strong>org.apache.sysml.test.integration.mlcontext.algorithms.MLContextLinregTest.testLinregDSDense</strong></a></li><li><a href='https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/2026/org.apache.systemml$systemml/testReport/org.apache.sysml.test.integration.mlcontext.algorithms/MLContextLinregTest/testLinregCGSparse/'><strong>org.apache.sysml.test.integration.mlcontext.algorithms.MLContextLinregTest.testLinregCGSparse</strong></a></li></ul>

**Comment by j143** [comment]
Hi @deroneriksson , this seem to have some validation and script execution errors, 
```
org.apache.sysml.api.mlcontext.MLContextException: Exception when executing script
	at org.apache.sysml.parser.Expression.raiseValidateError(Expression.java:532)
	at org.apache.sysml.parser.Expression.raiseValidateError(Expression.java:510)
	at org.apache.sysml.parser.DataExpression.validateExpression(DataExpression.java:610)
	at org.apache.sysml.parser.StatementBlock.validate(StatementBlock.java:597)
	at org.apache.sysml.parser.DMLTranslator.validateParseTree(DMLTranslator.java:158)
	at org.apache.sysml.api.mlcontext.ScriptExecutor.validateScript(ScriptExecutor.java:549)
	at org.apache.sysml.api.mlcontext.ScriptExecutor.compile(ScriptExecutor.java:309)
	at org.apache.sysml.api.mlcontext.ScriptExecutor.compile(ScriptExecutor.java:268)
	at org.apache.sysml.api.mlcontext.ScriptExecutor.execute(ScriptExecutor.java:345)
	at org.apache.sysml.api.mlcontext.MLContext.execute(MLContext.java:355)
	at org.apache.sysml.api.mlcontext.MLContext.execute(MLContext.java:332)
	at org.apache.sysml.test.integration.mlcontext.algorithms.MLContextLinregTest.runLinregTestMLC(MLContextLinregTest.java:74)
	at org.apache.sysml.test.integration.mlcontext.algorithms.MLContextLinregTest.testLinregCGDense(MLContextLinregTest.java:52)
```
Am I doing some mistake in giving inputs, I tried to go over the above pointed line numbers?. Thanks.

**Comment by deroneriksson** [comment]
Hi @j143. I believe there are a couple issues.

1) Looking at the LinearRegCG.dml script, it looks like the author of the LinearRegCG.dml script inconsistently made the input matrix 'y' lowercase (rather than uppercase, like 'X'). Therefore, `.in("Y", Y)` needs to be `.in("y", Y)`.
2) Secondly, there is no variable 'B' in the script, so defining 'B' as a variable that shall be output via `.out("B")` will fail. You could obtain 'beta_out', so `out("beta_out")` would work.

Also, note that no checks are being performed for expected results. I think that is probably fine at this stage, since these test demonstrate that these scripts can indeed be executed using the MLContext API.

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/2033/

**Comment by deroneriksson** [comment]
Thank you @j143. LGTM. Would you like to add further tests or can I merge this?

**Comment by j143** [comment]
If we don't need to worry about testing operation result through `MLContext`, 'cause the different operations are already tested normally and compared with `https://github.com/apache/systemml/blob/master/src/test/scripts/functions/codegen/Algorithm_LinregCG.R#L23`, we can go for merging. 

With the consequent algorithms I will be improving myself on how exactly the `mlcontext` functions. And if there is no comparison scripts, I'll try to add comparison `R` scripts.

**Comment by j143** [comment]
Also, I am unable update the jira issue status such as `progress`, `resolved` or `closed`. Please, allow me to make those changes for easy handling of issues. username: **@return_01**
Thanks.

File: src/test/java/org/apache/sysml/test/integration/mlcontext/algorithms/MLContextLinregTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.sysml.test.integration.mlcontext.algorithms;
+
+import static org.apache.sysml.api.mlcontext.ScriptFactory.dmlFromFile;
+
+import java.util.concurrent.ThreadLocalRandom;
```
Comment (j143) [inline src/test/java/org/apache/sysml/test/integration/mlcontext/algorithms/MLContextLinregTest.java]: Some of the imports were not used in the script. 
`import java.util.concurrent.ThreadLocalRandom;`

**Comment by deroneriksson** [comment]
@j143 I created an Apache Infrastructure to see if SystemML contributors can have access to the Start Progress/Stop Progress buttons. See https://issues.apache.org/jira/browse/INFRA-15023

**Comment by deroneriksson** [comment]
@j143 Yes, in the future, please remove unnecessary imports. I'll go ahead and remove them and merge. Thank you!

**Pull request closed without merging**